common/djangoapps/edxmako/management/commands/preprocess_assets.py

```diff
@@ -8,67 +8,53 @@
     """
 
     help = "Preprocess asset template files to ready them for compilation."
 
-    def handle_noargs(self, **options):
-        """
-        Walk over all of the static files directories specified in the
-        settings file, looking for asset template files (indicated by
-        a file extension like .mako).
-        """
-        for staticfiles_dir in getattr(settings, "STATICFILES_DIRS", []):
-            # Cribbed from the django-staticfiles app at:
-            # https://github.com/jezdez/django-staticfiles/blob/develop/staticfiles/finders.py#L52
-            if isinstance(staticfiles_dir, (list, tuple)):
-                prefix, staticfiles_dir = staticfiles_dir
-
-            # Walk over the current static files directory tree,
-            # preprocessing files that have a template extension.
-            for root, dirs, files in os.walk(staticfiles_dir):
-                for filename in files:
-                    outfile, extension = os.path.splitext(filename)
-                    # We currently only handle Mako templates
-                    if extension == ".mako":
-                        self.__preprocess(os.path.join(root, filename),
-                                          os.path.join(root, outfile))
-
-    def __context(self):
-        """
-        Return a dict that contains all of the available context
-        variables to the asset template.
-        """
-        # TODO: do we need to include anything else?
-        # TODO: do this with the django-settings-context-processor
-        return {
-            "FEATURES": settings.FEATURES,
-            "THEME_NAME": getattr(settings, "THEME_NAME", None),
-        }
-
-    def __preprocess(self, infile, outfile):
-        """
-        Run `infile` through the Mako template engine, storing the
-        result in `outfile`.
-        """
-        with open(outfile, "w") as _outfile:
-            _outfile.write(textwrap.dedent("""\
-            /*
-             * This file is dynamically generated and ignored by Git.
-             * DO NOT MAKE CHANGES HERE. Instead, go edit its template:
-             * %s
-             */
-            """ % infile))
-            _outfile.write(Template(filename=str(infile)).render(env=self.__context()))
+    def handle(self, *args, **options):
+        theme_name = getattr(settings, "THEME_NAME", None)
+        use_custom_theme = settings.FEATURES.get("USE_CUSTOM_THEME", False)
+        if not use_custom_theme or not theme_name:
+            # No custom theme, nothing to do!
+            return
+
+        dest_dir = args[-1]
+        for source_file in args[:-1]:
+            self.process_one_file(source_file, dest_dir, theme_name)
+
+    def process_one_file(self, source_file, dest_dir, theme_name):
+        """Pre-process a .scss file to replace our markers with real code."""
+        with open(source_file) as fsource:
+            original_content = content = fsource.read()
+
+        content = content.replace(
```

Comment:
Nit: I find this a bit confusing. `original_content = fsource.read()` and `content = original_content.replace(...)` would make more sense to me.